src/d3a/models/area/redis_dispatcher/market_event_dispatcher.py

```diff
@@ -0,0 +1,90 @@
+import json
+import logging
+from random import random
+from threading import Event
+from concurrent.futures import TimeoutError, ThreadPoolExecutor
+from d3a.events import MarketEvent
+from d3a.models.area.redis_dispatcher import RedisEventDispatcherBase
+from d3a.models.market.market_structures import parse_event_and_parameters_from_json_string
+
+
+class RedisMarketEventDispatcher(RedisEventDispatcherBase):
+    def __init__(self, area, root_dispatcher, redis):
+        super().__init__(area, root_dispatcher, redis)
+        self.str_market_events = [event.name.lower() for event in MarketEvent]
+        self.market_event = Event()
+        self.futures = []
+        self.executor = ThreadPoolExecutor(max_workers=10)
```

Comment:
Rule of thumb, this means that there can be processed up to 10 concurrent events. On large one-sided market simulations this could potentially be surpassed, therefore replacing this with a constant to adapt it in case we need more than 10 workers.